base_dj/models/base.py

```diff
@@ -197,27 +199,32 @@ def is_missing(r):
                 (record, to_xid(record.id))
                 for record in self
             )
-
         xids.update(
             (r.id, (modname, r._dj_xmlid_export_name())) for r in missing
         )
         # you can generate one shot xids and not store them
         # so you don't pollute your db and maybe fix some csv
         if not self.env.context.get('dj_xmlid_skip_create'):
             fields = ['module', 'model', 'name', 'res_id']
-            self.env.cr.copy_from(io.StringIO(
-                u'\n'.join(
-                    u"%s\t%s\t%s\t%d" % (
-                        modname,
-                        record._name,
-                        xids[record.id][1],
-                        record.id,
-                    )
-                    for record in missing
-                )),
-                table='ir_model_data',
-                columns=fields,
-            )
+            try:
+                self.env.cr.copy_from(io.StringIO(
+                    u'\n'.join(
+                        u"%s\t%s\t%s\t%d" % (
+                            modname,
+                            record._name,
+                            xids[record.id][1],
+                            record.id,
+                        )
+                        for record in missing
+                    )),
+                    table='ir_model_data',
+                    columns=fields,
+                )
+            except IntegrityError:
+                raise ValidationError(
+                    "Writing xmlids for %s failed."
```

Comment:
@tschanzt sorry, 1 more change: can you join the list of IDS here, maybe on a new line? It will ease debugging even if you won't know precisely which record failed if we have more than one...
2nd: can you upgrade module version?